src/schema/submissions.ts

```
@@ -0,0 +1,176 @@
+    """
+    Submit an article to surface on users feed
+    """
+    submitArticle(url: String!): SubmitArticle @auth
+  }
+`;
+
+const submissionLimit = parseInt(
+  process.env.SCOUT_SUBMISSION_LIMIT || DEFAULT_SUBMISSION_LIMIT,
+);
+
+export const submissionAccessThreshold = parseInt(
+  process.env.SCOUT_SUBMISSION_ACCESS_THRESHOLD ||
+    DEFAULT_SUBMISSION_ACCESS_THRESHOLD,
+);
+
+const hasSubmissionAccess = (user: User) =>
+  user.reputation >= submissionAccessThreshold;
+
+const getSubmissionsToday = (con: Connection, userTimezone?: string) => {
+  const timezone = userTimezone ?? 'utc';
+  const nowTimezone = `timezone('${timezone}', now())`;
+  const atTimezone = `at time zone '${timezone}'`;
+
+  return con.getRepository(Submission).find({
+    where: `date_trunc('day', "createdAt")::timestamptz ${atTimezone} = date_trunc('day', ${nowTimezone})::timestamptz`,
+  });
+};
+
+export const resolvers: IResolvers<unknown, Context> = traceResolvers({
+  Query: {
+    submissionAvailability: async (
+      _,
+      __,
+      ctx,
+    ): Promise<GQLSubmissionAvailability> => {
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+      if (!user) {
+        return {
+          limit: submissionLimit,
+          hasAccess: false,
+          todaySubmissionsCount: 0,
+        };
+      }
+
+      const submissionsToday = await getSubmissionsToday(
+        ctx.con,
+        user.timezone,
+      );
+
+      return {
+        limit: submissionLimit,
+        todaySubmissionsCount: submissionsToday.length,
+        hasAccess: hasSubmissionAccess(user),
+      };
+    },
+  },
+  Mutation: {
+    submitArticle: async (
+      _,
+      { url }: GQLArticleSubmission,
+      ctx,
+      info,
+    ): Promise<GQLSubmitArticleResponse> => {
+      const user = await ctx.getRepository(User).findOne({ id: ctx.userId });
+
+      if (!user || !hasSubmissionAccess(user)) {
+        return { result: 'reject', reason: 'Access denied' };
+      }
```

This should throw an error as we do for non-authenticated users